redwood-compose/src/commonTest/kotlin/app/cash/redwood/compose/BackHandlerTest.kt

```diff
@@ -0,0 +1,204 @@
+      assertThat(awaitSnapshot()).single().isEqualTo(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).single().isEqualTo(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostEnabledAndInnermostEnabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        var backCounter by remember { mutableStateOf(0) }
+        BackHandler(enabled = false, throwingOnBack)
+        Box {
+          BackHandler {
+            backCounter += 1
+          }
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostEnabledAndInnermostDisabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        var backCounter by remember { mutableStateOf(0) }
+        BackHandler {
+          backCounter += 1
+        }
+        Box {
+          BackHandler(enabled = false, throwingOnBack)
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostDisabledAndInnermostEnabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        var backCounter by remember { mutableStateOf(0) }
+        BackHandler(enabled = false, throwingOnBack)
+        Box {
+          BackHandler {
+            backCounter += 1
+          }
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostDisabledAndInnermostDisabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        val backCounter by remember { mutableStateOf(0) }
+        BackHandler(enabled = false, throwingOnBack)
+        Box {
+          BackHandler(enabled = false, throwingOnBack)
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertNoSnapshot()
+    }
+  }
+}
+
+private suspend fun TestRedwoodComposition<List<WidgetValue>>.assertNoSnapshot() {
+  assertFailure { awaitSnapshot() }.isInstanceOf<TimeoutCancellationException>()
+}
```

Comment:
I wonder if we can provide a proper API for this rather than relying on the timeout.

Reply:
Doesn't seem like it